bugsnag-android-core/src/main/java/com/bugsnag/android/Event.kt

```diff
@@ -0,0 +1,174 @@
+package com.bugsnag.android
+
+import java.io.IOException
+import java.util.HashMap
+
+/**
+ * Information and associated diagnostics relating to a handled or unhandled
+ * Exception.
+ *
+ * This object is made available in OnError callbacks, so you can
+ * inspect and modify it before it is delivered to Bugsnag.
+ *
+ * @see OnError
+ */
+class Event @JvmOverloads internal constructor(
+    val originalError: Throwable? = null,
+    internal val config: ImmutableConfig,
+    private var handledState: HandledState,
+    internal val metadata: Metadata = Metadata(),
+    threadState: ThreadState? = if (config.sendThreads) {
+        ThreadState(config, originalError)
+    } else null
+) : JsonStream.Streamable, MetadataAware {
+
+    var session: Session? = null
+
+    /**
+     * Set the Severity of this Event.
+     *
+     * By default, unhandled exceptions will be Severity.ERROR and handled
+     * exceptions sent with bugsnag.notify will be Severity.WARNING.
+     * @see Severity
+     */
+    var severity: Severity
+        get() = handledState.currentSeverity
+        set(value) {
+            handledState.currentSeverity = value
+        }
+
+    var apiKey: String = config.apiKey
+
+    var app: MutableMap<String, Any?> = HashMap()
+    var device: MutableMap<String, Any?> = HashMap()
+
+    val unhandled = handledState.isUnhandled
+
+    var breadcrumbs: List<Breadcrumb> = emptyList()
+
+    var projectPackages: Collection<String> = config.projectPackages
+    var errors: List<Error> = when (originalError) {
+        null -> listOf()
+        else -> Error.createError(originalError, projectPackages)
+    }
+
+    var threads: List<Thread> = threadState?.threads ?: listOf()
+
+    /**
+     * Get the grouping hash associated with this Event.
+     *
+     * @return the grouping hash, if set
+     */
+    /**
+     * Set a custom grouping hash to use when grouping this Event on the
+     * Bugsnag dashboard. By default, we use a combination of error class
+     * and top-most stacktrace line to calculate this, and we do not recommend
+     * you override this.
+     */
```

Comment:
I have resolved the conversion issues with the JavaDoc.

Generally speaking I have avoided adding new JavaDoc or making major changes to existing comments as I think we should be updating our docs website at the same time, and thinking a bit about sharing explanations for key concepts between platforms. For example, the concept of `groupingHash` is common across platforms so it would be feasible to share most of the documentation between different notifiers, and add extra information for any platform-specific details. Let me know if you'd like me to change this approach